overlord/ifacestate/handlers.go

```diff
@@ -303,3 +305,24 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 	setConns(st, conns)
 	return nil
 }
+
+func (m *InterfaceManager) doDiscardNamespace(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapSetup, err := snapstate.TaskSnapSetup(task)
+	if err != nil {
+		return err
+	}
+
+	snapName := snapSetup.Name()
+
+	// Run "snap-discard-ns $SNAP_NAME" and ignore the error code.  This
+	// command is not meant to always succeed because the namespace may not
+	// have been created yet in practice (e.g. the snap was removed before it
+	// was started since last reboot).
+	cmd := exec.Command(dirs.InternalCmdPath("snap-discard-ns"), snapName)
```

Comment:
Sure it is just more "magic" and less obvious what this means. I can change it if if you like though.